static/js/TextRange.jsx

```diff
@@ -1,41 +1,43 @@
-import React  from 'react';
+import React from 'react';
 import ReactDOM  from 'react-dom';
 import PropTypes  from 'prop-types';
 import classNames  from 'classnames';
 import $  from './sefaria/sefariaJquery';
 import Sefaria  from './sefaria/sefaria';
 import Component from 'react-class';
-import {EnglishText, HebrewText} from "./Misc";
+import {EnglishText, HebrewText, LoadingMessage} from "./Misc";
 import {VersionContent} from "./ContentText";
 import {ContentText} from "./ContentText";
+import {ReaderPanelContext} from "./context";
 
 class TextRange extends Component {
   // A Range or text defined a by a single Ref. Specially treated when set as 'basetext'.
   // This component is responsible for retrieving data from `Sefaria` for the ref that defines it.
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+      isMounted: false,
+    };
+  }
   componentDidMount() {
-    this._isMounted = true;
-    let data = this.getText();
-    if (data && !this.dataPrefetched) {
-      // If data was populated server side, onTextLoad was never called
-      this.onTextLoad(data);
-    } else if (this.props.basetext || this.props.segmentNumber) {
-      this.placeSegmentNumbers();
-    }
-    window.addEventListener('resize', this.handleResize);
+    this.setState({isMounted: true});
+    this.setData()
+    window.addEventListener('resize', this.conditionalPlaceSegmentNumbers);
   }
   componentWillUnmount() {
-    this._isMounted = false;
-    window.removeEventListener('resize', this.handleResize);
+    this.setState({isMounted: false});
+    window.removeEventListener('resize', this.conditionalPlaceSegmentNumbers);
```

Comment:
verify that changes don't stem from changing cache

Reply:
i don't see the connection to cache. there are 2 things here: 1. changing this._isMounted to isMounted that's stored in state, which i think is more reasonable. 2. changing the name of the function that place segments